iothub/device/src/TelemetryMethods.cs

```diff
@@ -29,8 +33,77 @@ public static string GetSqmMachineId()
                     if (Logging.IsEnabled) Logging.Error(null, ex, nameof(TelemetryMethods));
                 }
             }
+            else
+            {
+                try
+                {
+                    // get HMAC-SHA256 of the ADHS application ID, keyed by the machine ID
+                    using var sr = new StreamReader("/etc/machine-id");
+                    string key = sr.ReadLine();
+                    if (!string.IsNullOrEmpty(key))
+                    {
+                        // ADHS application ID = 586e0ed989c946a794e5de337644d308
+                        return GetUUIDv4("586e0ed989c946a794e5de337644d308", key);
```

Comment:
Can you also point to public doc describing this ID?

Reply:
The ADHS application ID = 586e0ed989c946a794e5de337644d308 is a hardcoded value ([here](https://mscodehub.visualstudio.com/Azile/_git/azure-device-health-services?path=%2Fdevice-utils%2Fsrc%2FIdentity.cpp&_a=contents&version=GBmaster)) used by the ADHS client to derive a "device id" with which all telemetry data from a device is tamped with. The "device id" is practically a hash (HMAC-SHA256) of the ADHS application ID, keyed by the machine ID. The latter is a secret that should never be exposed.
ADHS is expected to go public as open-source in the next semester (Cobalt).

Reply:
I'm concerned that without public documentation; this would be something hard to explain/maintain as we are open source. Also I don't see "The "device id" is practically a hash (HMAC-SHA256) of the ADHS application ID, keyed by the machine ID. The latter is a secret that should never be exposed." as an iot SDK concept. So why is this sitting in SDK?